### PR TITLE
Retry Test Airflow Release Step Multiple Times 

### LINKED ISSUE #42025: Test Airflow Release step is flaky.
The "Test Airflow Releases" job is flaky and fails far too often. 

Most of this comes from "external" factors - for example installing node packages, pulling images etc. often fail with 500 internal error or "Rate limit exceeded" . 

Example https://github.com/apache/airflow/actions/runs/10717569021/job/29717755211?pr=41555  where root cause is installing node packages:

```
 yarn install v1.22.21
  (node:499) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
  (Use `node --trace-deprecation ...` to show where the warning was created)
  [1/4] Resolving packages...
  [2/4] Fetching packages...
  [] 0/1573[] 7/1573[] 13/1573[] 21/1573[] 29/1573error Error: https://registry.yarnpkg.com/@chakra-ui/skeleton/-/skeleton-2.0.18.tgz: Request failed "500 Internal Server Error"
      at ResponseError.ExtendableBuiltin (/opt/airflow/files/home/.cache/pre-commit/repoj5n0lz2l/node_env-22.2.0/lib/node_modules/yarn/lib/cli.js:696:66)
      at new ResponseError (/opt/airflow/files/home/.cache/pre-commit/repoj5n0lz2l/node_env-22.2.0/lib/node_modules/yarn/lib/cli.js:802:124)
      at Request.<anonymous> (/opt/airflow/files/home/.cache/pre-commit/repoj5n0lz2l/node_env-22.2.0/lib/node_modules/yarn/lib/cli.js:66218:16)
      at Request.emit (node:events:520:28)
      at module.exports.Request.onRequestResponse (/opt/airflow/files/home/.cache/pre-commit/repoj5n0lz2l/node_env-22.2.0/lib/node_modules/yarn/lib/cli.js:141751:10)
      at ClientRequest.emit (node:events:520:28)
      at HTTPParser.parserOnIncomingClient (node:_http_client:700:27)
      at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
      at TLSSocket.socketOnData (node:_http_client:542:22)
      at TLSSocket.emit (node:events:520:28)
  info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
  Traceback (most recent call last):
    File "./scripts/ci/pre_commit/compile_www_assets.py", line 71, in <module>
      subprocess.check_call(["yarn", "install", "--frozen-lockfile"], cwd=os.fspath(www_directory))
    File "/usr/local/lib/python3.8/subprocess.py", line 364, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['yarn', 'install', '--frozen-lockfile']' returned non-zero exit status 1.
```

The solution to that is likely attempting to retry the whole `breeze` command several times. This "release process" is relatively fast (~4 minutes) so retrying it up to 3 times in case of failures will bring the total time to 12 minutes max and should not have much impact on elapsed time or cost.

### PR DESCRIPTION
**Pull request by bugraoz93**
closes: #42025
* Script added for retrying non-interactive commands.
* The script is used for retrying breeze commands under `Test Airflow Release Step` 3 times

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by bugraoz93** [comment]
The script including retrying the resulting permission denied. I have checked the other scripts such as `install_breeze.sh` but there wasn't any special case. I have started thinking I don't have enough permissions to include custom scripts for protection.

**Comment by potiuk** [comment]
It's likely about executable permission for the script

**Comment by potiuk** [comment]
`chnod a+x script.sh`

**Comment by bugraoz93** [comment]
I thought the same at the beginning either executable permissions or ownership of the file was not exactly correct, I checked how you included the `scripts/ci/testing/run_integration_tests_with_retry.sh` file into the `integration-tests` since the approach should be pretty much the same. #39913 In this one, I couldn't find any trace of `chmod` included for the script somewhere in the project and the `integration-tests` step worked. So I assumed the scripts were included in the CI image without specifying permission of the file. I may be missing something :thinking:

**Comment by potiuk** [comment]
> So I assumed the scripts were included in the CI image without specifying permission of the file. I may be missing something 

There is no "trace" when you look at the file "content" - you need to see at the git file status a bit mre deeply. See for example accepted answer in https://stackoverflow.com/questions/46175753/how-do-i-see-the-permission-of-a-file-in-git - you need to manually set "chmod a+x" on the file, then add the file again and you will see that permissions of the file in git are diferent than the permission of other files:

You can also see it locally. When I check out your PR, I see:

```
[jarek:~/code/airflow] retry-test-Airflow-release-step+ ± ls -la scripts/ci/testing
total 48
drwxr-xr-x   6 jarek  staff    192 Sep  8 04:45 .
drwxr-xr-x  15 jarek  staff    480 Aug 27 14:13 ..
-rw-r--r--   1 jarek  staff   2258 Sep  8 04:45 run_breeze_command_with_retries.sh
-rwxr-xr-x   1 jarek  staff   1719 Aug 27 14:13 run_integration_tests_with_retry.sh
-rwxr-xr-x   1 jarek  staff  10839 Aug 27 14:13 summarize_captured_warnings.py
-rwxr-xr-x   1 jarek  staff   3859 Aug 27 14:13 summarize_junit_failures.py
```

as you see - comparing to other scripts in the folder, your script misses the "executable" bit - and the problem is that this executable bit is not stored in `git`.


So in this case:

```bash
chmod a+x ./scripts/testing/run_breeze_command_with_retries.sh
git add ./scripts/testing/run_breeze_command_with_retries.sh
git commit --amend
```

should do the job.

You will see after `git add` that even if the content of the file did not change, the "executable bit" did - and git wil offer you to modify the executable bit for that file when you commit it.

**Comment by bugraoz93** [comment]
Thanks for the detailed answer! I thought this was somehow an automated process. How should I miss that :facepalm:

File: scripts/ci/testing/run_breeze_command_with_retries.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ATTEMPT_SUCCESS=false
+
+breeze down
+set +e
+CURRENT_RUN_OUTPUT=$($COMMAND_TO_ATTEMPT)
+RESULT=$?
+echo "$CURRENT_RUN_OUTPUT"
+set -e
+
+while [ "${ATTEMPT_SUCCESS}" = false ] && [ "${CURRENT_ATTEMPT}" -le "${NUMBER_OF_ATTEMPT}" ]; do
```
Comment (potiuk) [inline scripts/ci/testing/run_breeze_command_with_retries.sh]: I think this script might be vastly simplified (many less environment variables and simpler steps) by using for loop and passing all arguments of the script to the breeze command:

Something like that (with added diagnostics and printing of course) should ldo the job - might need some tweaking, as I am writing it from the top of my head :) :

```bash
for i in $(seq 1 "${NUM_ATTEMPTS}")
do
   if breeze "$@" then
      # success here
      exit 0
   else
     RESULT=$?
     # restart here (maybe optimized by exit early `if "$i" == "${NUM_ATTEMPTS}"`
   fi
done

# Error here
```

Comment (bugraoz93) [reply scripts/ci/testing/run_breeze_command_with_retries.sh]: Make sense! Let me apply the logic to the script. Thanks :)

**Review by potiuk — APPROVED** [review]
This looks cool !

**Comment by bugraoz93** [comment]
Awesome! Many thanks!

**Pull request merged**